pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -147,6 +148,39 @@ func StartReplicationProducerJob(
 	}, nil
 }
 
+// StartHistoryProtectionJob replication stream producer job
+// responsible for managing a cluster-level protected timestamp.
+func StartHistoryProtectionJob(
```

Comment:
We can avoid alot of the indirection here if we are OK moving the producer job resumer to `pkg/repstream`. That would make it available in non-CCL binaries. Then we would move StartHistoryProtectionJob to just live on sql.Planner. If this seems cool with y'all let me know.

I don't think the job's resumer particularly needs to be restricted to CCL given that we would still be restricting many of the builtins to CCL code.